mne/io/brainvision/brainvision.py

```diff
@@ -502,29 +502,30 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
                 # highpass relaxed / no filters
                 highpass = [float(filt) if filt not in ('NaN', 'Off', 'DC')
                             else np.Inf for filt in highpass]
-                info['highpass'] = np.min(np.array(highpass, dtype=np.float))
+                info['highpass'] = np.max(np.array(highpass, dtype=np.float))
                 # Coveniently enough 1 / np.Inf = 0.0, so this works for
                 # DC / no highpass filter
                 info['highpass'] = 1. / info['highpass']
 
                 # not exactly the cleanest use of FP, but this makes us
                 # more conservative in *not* warning.
-                if info['highpass'] == 0.0:
+                if info['highpass'] == 0.0 and len(set(highpass)) == 1:
                     # not actually heterogeneous in effect
                     # ... just heterogeneously disabled
                     heterogeneous_hp_filter = False
             else:
                 highpass = [float(filt) if filt not in ('NaN', 'Off', 'DC')
                             else 0.0 for filt in highpass]
-                info['highpass'] = np.max(np.array(highpass, dtype=np.float))
-                if info['highpass'] == 0.0:
+                info['highpass'] = np.min(np.array(highpass, dtype=np.float))
+                if info['highpass'] == 0.0 and len(set(highpass)) == 1:
                     # not actually heterogeneous in effect
                     # ... just heterogeneously disabled
                     heterogeneous_hp_filter = False
+
             if heterogeneous_hp_filter:
                 warn('Channels contain different highpass filters. '
-                     'Highest filter setting (%0.2f Hz) will be stored.'
-                     % info['highpass'])
+                     'Lowest (weakest) filter setting (%0.2f Hz) '
+                     'will be stored.' % info['highpass'])
```

Comment:
Highest in seconds is lowest in Hertz; both the Hertz and the seconds case should be covered by tests.